src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -57,24 +57,33 @@ public static string[] GetCoverableModules(string module, string[] directories)
 
         public static bool HasPdb(string module)
         {
-            using (var moduleStream = File.OpenRead(module))
-            using (var peReader = new PEReader(moduleStream))
+            try
             {
-                foreach (var entry in peReader.ReadDebugDirectory())
+                using (var moduleStream = File.OpenRead(module))
+                using (var peReader = new PEReader(moduleStream))
                 {
-                    if (entry.Type == DebugDirectoryEntryType.CodeView)
+                    foreach (var entry in peReader.ReadDebugDirectory())
                     {
-                        var codeViewData = peReader.ReadCodeViewDebugDirectoryData(entry);
-                        if (codeViewData.Path == $"{Path.GetFileNameWithoutExtension(module)}.pdb")
+                        if (entry.Type == DebugDirectoryEntryType.CodeView)
                         {
-                            // PDB is embedded
-                            return true;
+                            var codeViewData = peReader.ReadCodeViewDebugDirectoryData(entry);
+                            if (codeViewData.Path == $"{Path.GetFileNameWithoutExtension(module)}.pdb")
+                            {
+                                // PDB is embedded
+                                return true;
+                            }
+
+                            return File.Exists(codeViewData.Path);
                         }
-
-                        return File.Exists(codeViewData.Path);
                     }
-                }
 
+                    return false;
+                }
+            }
+            catch (Exception ex)
```

Comment:
I think it's better try/catch where we use HasPdb https://github.com/tonerdo/coverlet/blob/13c069cde013d6cf63c90a91a9f19588b0d37ea7/src/coverlet.core/Instrumentation/Instrumenter.cs#L53
and use logger 
https://github.com/tonerdo/coverlet/blob/13c069cde013d6cf63c90a91a9f19588b0d37ea7/src/coverlet.core/Instrumentation/Instrumenter.cs#L29

https://github.com/tonerdo/coverlet/blob/f25833315cd2f0a124330754dc85f01a16ce6efa/src/coverlet.msbuild.tasks/InstrumentationTask.cs#L102

Reply:
Ok